### PR TITLE
home: create page

### PR DESCRIPTION
**Pull request by mt-bt**
Issue #11

### CODE REVIEW

File: src/cljs/pancetta/components/home.cljs [hunk 1]
```diff
@@ -1,23 +1,86 @@
...
+   [:span (str " for " (price-to-str (:price ticket)))]])
+
+(defn form-component [state]
+  [:div {:class-name "pure-u-3-5"}
+    [:form {:style (:form-container style) :class-name "pure-form"}
+     [:h2 "Find a ticket"]
+     [:div {:style (:form-group style)}
+      [:span "I am looking for a train on the "]
+      [:input {:type "date"
+               :min (-> (js/Date.) .toISOString (.slice 0 10))
```
Comment (teawaterwire) [inline src/cljs/pancetta/components/home.cljs:54]: Couldn't re-use the date-picker here?

Comment (mt-bt) [inline src/cljs/pancetta/components/home.cljs:54]: I didn't know you created a component for that. 

EDIT: It's not actually a component yet. I suggest we create a proper date picker later on, but we can merge that for now, don't you think?

File: src/cljs/pancetta/components/home.cljs [hunk 2]
```diff
@@ -1,23 +1,86 @@
...
+                 :default-value "Paris"}
+       (for [city (doall (filter #(not= % (:from @query)) cities))]
+         [:option {:key city} city])]]
+     [:a {:on-click #(get-tickets (:root @state))
+          :class-name "button-primary pure-button"} "Search"]]])
+
+(defn results-component [state]
+  [:div {:class-name "pure-u-2-5"}
+   [:div {:style (:results-container style)}
+    (if (nil? @tickets)
```
Comment (teawaterwire) [inline src/cljs/pancetta/components/home.cljs:76]: do we need those nested `if` ?

Comment (mt-bt) [inline src/cljs/pancetta/components/home.cljs:76]: yes we do

File: src/cljs/pancetta/components/home.cljs (hunk 1 above)
Comment (teawaterwire) [inline src/cljs/pancetta/components/home.cljs:54]: yep we don't want early optimisations :dancer:

**Comment by mt-bt** [comment]
?

File: src/cljs/pancetta/components/home.cljs [hunk 3]
```diff
@@ -1,23 +1,86 @@
...
+                             :margin 5}
+            :results-container {:margin 5}
+            :ticket {:background-color "white"
+                     :margin 1
+                     :border-radius 3
+                     :padding 10
+                     :cursor "pointer"}
+            :form-group {:margin "20px 0px"}})
+
+(def query (atom {:from "London"
```
Comment (teawaterwire) [inline src/cljs/pancetta/components/home.cljs]: `defonce` for figwheel goodness

File: src/cljs/pancetta/components/home.cljs [hunk 4]
```diff
@@ -1,23 +1,86 @@
...
+                  :to "Paris"
+                  :on (-> (js/Date.) .toISOString (.slice 0 10))}))
+
+(def tickets (atom nil))
```
Comment (teawaterwire) [inline src/cljs/pancetta/components/home.cljs]: same here

File: src/cljs/pancetta/components/home.cljs [hunk 5]
```diff
@@ -1,23 +1,86 @@
...
-              (get-in ticket [:price :amount])])]])))
+                     (m/get-in root [:tickets])
+                     (.orderByChild "on")
+                     (.equalTo (:on @query)))]
+    (m/listen-to tickets-ref
+                 :value
+                 (fn [[id res]] (reset! tickets
+                                        (doall (filter
+                                                (fn [[id ticket]] (and (= (:to @query) (:to ticket))
+                                                                       (= (:from @query) (:from ticket)))) res)))))))
```
Comment (teawaterwire) [inline src/cljs/pancetta/components/home.cljs]: Alternative for the callback `fn`:

``` clj
(fn [[_ res]]
                   (reset! tickets (for [[_ {:keys [to from]} :as ticket] res
                                         :when (and (= (:to @query) to)
                                                    (= (:from @query) from))]
                                     ticket)))
```

Comment (mt-bt) [inline src/cljs/pancetta/components/home.cljs]: yep nice, sorry I am quite busy right now, if you need feel free to update the PR yourself.

**Comment by teawaterwire** [comment]
Merging it. I agree the best thing would be to have a matrix view to start

**Comment by mt-bt** [comment]
It would be great, are you working on that now?

**Comment by teawaterwire** [comment]
I'll create an issue for that

**Pull request merged**